hazelcast/src/test/java/com/hazelcast/sql/impl/expression/datetime/FormatterTest.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.datetime;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.ZoneOffset;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:ParenPad")
+public class FormatterTest {
```

Comment:
This test fails:
```java
    @Test
    public void testBasic() {
        Formatter f = new Formatter("9");
        check(0, f, " 0");
        f = new Formatter("FM9");
        check(0, f, "0");
    }
```

Reply:
```
@Test
public void testBasic() {
    Formatter f = new Formatter("0");
    check(0, f, " 0");
    f = new Formatter("FM0");
    check(0, f, "0");
}
```

Reply:
We should add this test to our tests, we don't have a similar one.

Reply:
This no longer fails, but we don't cover it in tests.

Reply:
We cover `9` and `FM99`.